### PR TITLE
Add section on dbt provider to dbt guide

### PR DESCRIPTION
**Pull request by kentdanas**
Added section on dbt Cloud that discusses the new dbt provider. Minor updates throughout the rest to distinguish Cloud vs Core and remove references to outdates community packages. @josh-fell if you could review and make sure I got the gist right that would be much appreciated!

### CODE REVIEW

File: guides/airflow-dbt.md [hunk 1]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
 
 Teams can use Airflow to orchestrate and execute dbt models as DAGs. Running dbt with Airflow ensures a reliable, scalable environment for models, as well as the ability to trigger models only after every prerequisite task is met. Airflow also gives you fine-grained control over dbt tasks such that teams have observability over every step in their dbt models.
 
-In this guide, we'll walk through two common use cases for orchestrating dbt with Airflow via the `BashOperator`: one at the [project](https://docs.getdbt.com/docs/building-a-dbt-project/projects) level, and another at the [model](https://docs.getdbt.com/docs/building-a-dbt-project/building-models) level. We'll also walk through how to extend the model-level use case by automating changes to a dbt model.
+In this guide, we'll walk through:
+
+- Using the [dbt Provider](https://registry.astronomer.io/providers/dbt-cloud) to orchestrate dbt Cloud with Airflow.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:18]:
```suggestion
- Using the [dbt Cloud Provider](https://registry.astronomer.io/providers/dbt-cloud) to orchestrate dbt Cloud with Airflow.
```

File: guides/airflow-dbt.md [hunk 2]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+- Two common use cases for orchestrating dbt Core with Airflow via the `BashOperator`: 
+    - At the [project](https://docs.getdbt.com/docs/building-a-dbt-project/projects) level and, 
+    - At the [model](https://docs.getdbt.com/docs/building-a-dbt-project/building-models) level. 
+- How to extend the model-level use case by automating changes to a dbt model.
+
+> The dbt Core sections of this guide summarize some of the key practices and findings from our blog series with [Updater](https://updater.com/) and [Sam Bail](https://sambail.com/) about using dbt in Airflow. For more information, check out [Part 1](https://www.astronomer.io/blog/airflow-dbt-1), [Part 2](https://www.astronomer.io/blog/airflow-dbt-2), and [Part 3](https://www.astronomer.io/blog/airflow-dbt-3) of the series.
+
+## dbt Cloud
+
+To orchestrate [dbt Cloud](https://www.getdbt.com/product/what-is-dbt/) jobs with Airflow, you can use the [dbt Provider](https://registry.astronomer.io/providers/dbt-cloud), which contains the following useful modules:
```
Comment (josh-fell) [inline guides/airflow-dbt.md:28]:
```suggestion
To orchestrate [dbt Cloud](https://www.getdbt.com/product/what-is-dbt/) jobs with Airflow, you can use the [dbt Cloud Provider](https://registry.astronomer.io/providers/dbt-cloud), which contains the following useful modules:
```

File: guides/airflow-dbt.md [hunk 3]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+
+- **`DbtCloudRunJobOperator`:** Executes a dbt Cloud job.
+- **`DbtCloudGetJonRunArtifactOperator`:** Downloads artifacts from a dbt Cloud job run.
+- **`DbtCloudJobRunSensor`:** Waits for a dbt Cloud job run to complete.
+
+In order to use the dbt Provider in your DAGs, you will need to complete the following steps:
```
Comment (josh-fell) [inline guides/airflow-dbt.md:34]:
```suggestion
In order to use the dbt Cloud Provider in your DAGs, you will need to complete the following steps:
```

File: guides/airflow-dbt.md [hunk 4]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+
+1. Add the `apache-airflow-providers-dbt-cloud` package to your Airflow environment. If you are working in an Astro project, you can add the package to your `requirements.txt` file.
+2. Set up an Airflow connection to your dbt Cloud instance. The connection type should be `dbt Cloud`, and it should include an API token from your dbt Cloud account. If you want your dbt Provider tasks to use a default account ID, you can add that to the connection, but it is not required.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:37]:
```suggestion
2. Set up an Airflow connection to your dbt Cloud instance. The connection type should be `dbt Cloud`, and it should include an API token from your dbt Cloud account. If you want your dbt Cloud Provider tasks to use a default account ID, you can add that to the connection, but it is not required.
```

Comment (josh-fell) [reply guides/airflow-dbt.md:37]: Should also be a link to the [connection doc](https://airflow.apache.org/docs/apache-airflow-providers-dbt-cloud/stable/connections.html) in Airflow here?

File: guides/airflow-dbt.md [hunk 5]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+
+In the DAG below, we show a simple implementation of the dbt Provider. This example showcases how to run a dbt Cloud job from Airflow, while adding an operational check to ensure the dbt Cloud job is not running prior to triggering. The `DbtCloudHook` provides a `list_job_runs()` method which can be used to retrieve the latest triggered run for a job and check the status of said run. If the job is not in a state of 10 (Success), 20 (Error), or 30 (Cancelled), the pipeline will not try to trigger it.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:39]:
```suggestion
In the DAG below, we show a simple implementation of the dbt Cloud Provider. This example showcases how to run a dbt Cloud job from Airflow, while adding an operational check to ensure the dbt Cloud job is not running prior to triggering. The `DbtCloudHook` provides a `list_job_runs()` method which can be used to retrieve all runs for a given job. The operational check uses this method to retrieve the latest triggered run for a job and check its status. If the job is currenly not in a state of 10 (Success), 20 (Error), or 30 (Cancelled), the pipeline will not try to trigger another run.
```

Comment (josh-fell) [reply guides/airflow-dbt.md:39]: >If the job is not in a state of 10 (Success), 20 (Error), or 30 (Cancelled), the pipeline will not try to trigger it.

This unfortunately will all change when V4 of the dbt Cloud API is released and the provider is updated. Allegedly happening some time this year. Just a preemptive warning.

File: guides/airflow-dbt.md [hunk 6]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+    )
+
+    begin >> check_job >> Label("Job not currently running. Proceeding.") >> trigger_job >> end
+
+dag = check_before_running_dbt_cloud_job()
+```
+
+Note that in the `DbtCloudRunJobOperator` you must provide the dbt connection ID as well as the `job_id` of the job you are triggering.
+
+> **Note:** The full code for this example, along with other DAGs that implement the dbt Provider, can be found on the [Astronomer Registry](https://registry.astronomer.io/dags?providers=dbt+Cloud&page=1).
```
Comment (josh-fell) [inline guides/airflow-dbt.md:95]:
```suggestion
> **Note:** The full code for this example, along with other DAGs that implement the dbt Cloud Provider, can be found on the [Astronomer Registry](https://registry.astronomer.io/dags?providers=dbt+Cloud&page=1).
```

File: guides/airflow-dbt.md [hunk 7]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+        check_interval=600,
+        timeout=3600,
+    )
+
+    begin >> check_job >> Label("Job not currently running. Proceeding.") >> trigger_job >> end
+
+dag = check_before_running_dbt_cloud_job()
+```
+
+Note that in the `DbtCloudRunJobOperator` you must provide the dbt connection ID as well as the `job_id` of the job you are triggering.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:93]: Technically dbt Cloud connection ID doesn't need to passed in directly if folks use the default conn ID, but maybe this is not a specific that we typically caveat in the guides?

File: guides/airflow-dbt.md [hunk 8]
```diff
@@ -9,21 +9,96 @@ tags: ["DAGs", "Integrations"]
...
+
+> **Note:** The full code for this example, along with other DAGs that implement the dbt Provider, can be found on the [Astronomer Registry](https://registry.astronomer.io/dags?providers=dbt+Cloud&page=1).
 
-## Use Case 1: dbt + Airflow at the Project Level
+## dbt Core
 
-When orchestrating dbt, we have a few options for our DAGs. We can use:
+When orchestrating dbt Core with Airflow, the most straight forward DAG design is to run dbt commands directly through the [`BashOperator`](https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/bash.html). In this section we'll show a couple examples for how to do so.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:99]: >... the most straight forward DAG design

Maybe phrase it as "A straightforward DAG design"?

Comment (josh-fell) [reply guides/airflow-dbt.md:99]: Should we link to the `BashOperator` page in the Registry instead?

File: guides/airflow-dbt.md (hunk 5 above)
Comment (josh-fell) [reply guides/airflow-dbt.md:39]: Should the `DbtCloudHook` be listed as a "useful module" as well above?

**Review by josh-fell — CHANGES_REQUESTED** [review]
Looks smashing! I feel like we should be really consistent when saying "dbt Cloud Provider" instead of "dbt Provider" _just in case_ someone has a weird idea of trying to add a dbt Core provider to Airflow.  A few other suggestions but nothing major or required really.

File: guides/airflow-dbt.md (hunk 4 above)
Comment (kentdanas) [reply guides/airflow-dbt.md:37]: Sure, can add one

File: guides/airflow-dbt.md (hunk 7 above)
Comment (kentdanas) [reply guides/airflow-dbt.md:93]: Yeah I think better to explicitly show it being passed in so people realize a connection is needed. Won't hurt anything if they pass the default in, even if it's not necessary.

File: guides/airflow-dbt.md (hunk 8 above)
Comment (kentdanas) [reply guides/airflow-dbt.md:99]: Yes, good call

File: guides/airflow-dbt.md [hunk 9]
```diff
@@ -9,21 +9,97 @@ tags: ["DAGs", "Integrations"]
...
+
+Note that in the `DbtCloudRunJobOperator` you must provide the dbt connection ID as well as the `job_id` of the job you are triggering.
+
+> **Note:** The full code for this example, along with other DAGs that implement the dbt Cloud Provider, can be found on the [Astronomer Registry](https://registry.astronomer.io/dags?providers=dbt+Cloud&page=1).
 
-## Use Case 1: dbt + Airflow at the Project Level
+## dbt Core
 
-When orchestrating dbt, we have a few options for our DAGs. We can use:
+When orchestrating dbt Core with Airflow, a straight forward DAG design is to run dbt commands directly through the [`BashOperator`](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator). In this section we'll show a couple examples for how to do so.
```
Comment (josh-fell) [inline guides/airflow-dbt.md:100]:
```suggestion
When orchestrating dbt Core with Airflow, a straightforward DAG design is to run dbt commands directly through the [`BashOperator`](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator). In this section we'll show a couple examples for how to do so.
```

File: guides/airflow-dbt.md [hunk 10]
```diff
@@ -369,4 +445,4 @@ With regards to the `dbt test` runs:
 
 ## Conclusion
 
-To recap, in this guide we have learned about dbt, how to create dbt tasks in Airflow, and how to productionize those tasks to automatically create tasks based on a manifest. For a more detailed discussion on trade-offs, limitations, and adding dbt to a full ELT pipeline, see our blog posts. To see more examples of how to use dbt and Airflow to build pipelines, check out our [dbt DAGs on the Registry](https://registry.astronomer.io/dags/?query=dbt&badges=certified).
+To recap, in this guide we have learned about dbt, how to create dbt tasks in Airflow, and how to productionize those tasks to automatically create tasks based on a manifest. For a more detailed discussion on trade-offs, limitations, and adding dbt Core to a full ELT pipeline, see our blog posts. To see more examples of how to use dbt and Airflow to build pipelines, check out our [dbt DAGs on the Registry](https://registry.astronomer.io/dags/?query=dbt&badges=certified).
```
Comment (josh-fell) [inline guides/airflow-dbt.md:448]:
```suggestion
To recap, in this guide we have learned about dbt Cloud and dbt Core, how to create and productionize dbt tasks in Airflow, and how to automatically create dbt Core tasks based on a manifest. For a more detailed discussion on trade-offs, limitations, and adding dbt Core or dbt Cloud to a full ELT pipeline, see our blog posts. To see more examples of how to use dbt and Airflow to build pipelines, check out our [dbt DAGs on the Registry](https://registry.astronomer.io/dags/?query=dbt&badges=certified).
```

**Review by josh-fell — APPROVED** [review]
Just two tiny things but otherwise LGTM. 🚀

**Pull request merged**